### PR TITLE
Adding access-control-allow-headers

### PR DESCRIPTION
**Pull request by dalanmiller**
Should satisfy #765 let's see if tests pass...

Do you think you could give this a try @hoducha? 

i.e. https://discuss.horizon.io/t/problem-with-enabling-cors/472

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/819)

### CODE REVIEW

File: cli/src/serve.js [hunk 1]
```diff
@@ -335,6 +340,7 @@ const start_horizon_server = (http_servers, opts) =>
     permissions: opts.permissions,
     project_name: opts.project_name,
     access_control_allow_origin: opts.access_control_allow_origin,
+    access_control_allow_headers: opts.access_control_allow_headers
```
Comment (deontologician) [inline cli/src/serve.js]: missing comma at the end of the line

File: server/src/server.js [hunk 2]
```diff
@@ -152,6 +152,7 @@ class Server {
         res.writeHead(200, {
           'Content-Type': 'application/json',
           'Access-Control-Allow-Origin': opts.access_control_allow_origin,
+          'Access-Control-Allow-Headers': opts.access_control_allow_headers,
```
Comment (deontologician) [inline server/src/server.js:156]: Only add if option is not false/empty

**Comment by hoducha** [comment]
@dalanmiller: I had given it a try and it worked after fixing the issue that @deontologician mentioned above. However after authenticated using Google, it redirects to the Horizon server https://localhost:8181/?horizon_token=... while I think it should redirect to the app server so that the token can be stored on the client side. Please see my setup here: https://discuss.horizon.io/t/problem-with-enabling-cors/472. Thanks.

**Comment by hoducha** [comment]
@dalanmiller: Since it's almost done, could you please fix the small issues and merge it to the main branch?

**Comment by Adam-Burke** [comment]
How is this fix coming along? I can't progress past the X-Requested-With cors error.

**Comment by madwill** [comment]
Poking this issue! Would love to have Cors added for development !

**Comment by jirikolarik** [comment]
Can this be merged?

**Comment by theorye** [comment]
I neeeed this fix.

**Comment by dalanmiller** [comment]
@marshall007 are merge permissions still limited to former RethinkDB folk?

**Comment by theorye** [comment]
@dalanmiller I believe last pass expired.

**Comment by atomicleopard** [comment]
I'm trying to serve content statically outside of horizons, but use the horizons OAuth mechanism. Having CORS would permit the OAuth calls to work - so having this merged would be great. However the OAuth redirect does not return back to my client, but rather the horizons static location, so this is only half the story in this use case.

**Comment by riyadhzen** [comment]
@atomicleopard I didn't get your point fully. When you say that you are trying to serve content statically outside of horizon, do you mean something like fetching images from a CDN or am I missing something.

**Comment by atomicleopard** [comment]
@riyadhzen I want to host my web application in something like GCS/S3 to decouple SPA release cycles from horizons deployments. In doing so, in local dev I'm hosting on a different port, and in production possibly on a different subdomain. horizon.authEndpoint('github') (which fails for CORS reasons listed here) returns a location, which the browser redirects to (say, google or FB) - the redirect url passed to these returned the user to the horizons application, not the requesting front-end.

So I want localhost:3000 -> Google -> localhost:3000, 
but I get localhost:3000 -> Google -> localhost:8181

Hope that helps explain.

**Comment by riyadhzen** [comment]
Got it, thanks.

**Comment by danielpiedra** [comment]
This fix would make such a big thing, many of us are stuck at this point. anyone knows if this merge will be soon released?

**Pull request closed without merging**